lab/homework/index.js

```diff
@@ -0,0 +1,148 @@
+export class HomeworkError extends Error {
+    
+    static ERROR_CODE = {
+        NOT_A_PRODUCT: 'NOT A PRODUCT',
+        NOT_IN_CART: 'NOT IN CART',
+        EXPIRED: 'EXPIRED',
+        POOR_ALERT: 'NO MONEY'
+    };
+
+    static ERROR_MESSAGE = {
+        [HomeworkError.ERROR_CODE.NOT_A_PRODUCT]: 'The method must receive an instance of product!',
+        [HomeworkError.ERROR_CODE.NOT_IN_CART]: 'The product must exist in the cart!',
+        [HomeworkError.ERROR_CODE.EXPIRED]: 'The aliment is expired!',
+        [HomeworkError.ERROR_CODE.POOR_ALERT]: 'The customer does not have enough money!',
+    };
+    
+    constructor({ code, message = HomeworkError.ERROR_MESSAGE[code] }) {
+        super(message);
+        this.code = code;
+    }
+    
+    toString() {
+        return this.message + '\n' + this.stack;
+    }
+    
+}
+
+
+
+export class Product {
+    
+    constructor({name, price}) {
+        this.name = name;
+        this.price = price;
+    }
+    
+}
+
+
+
+export class Aliment extends Product {
+
+    constructor({name, price, expirationDate, grammage}) {
+        super({name, price});
+        this.expirationDate = expirationDate;
+        this.grammage = grammage;
+    }
+    
+    isExpired() {
+        // compare expiration date with current date
+        return this.expirationDate < new Date();
+    }
+
+}
+
+
+
+export class Furniture extends Product {
+
+    constructor({name, price, colour}) {
+        super({name, price});
+        this.colour = colour;
+    }
+
+}
+
+
+
+export class Person {
+    
+}
+
+
+
+export class Customer extends Person {
+    
+    constructor({money, shoppingCart}) {
+        super({});
+        this.money = money;
+        this.shoppingCart = shoppingCart;
+    }
+    
+    addToCart(product) {
+        if (product instanceof Product) {
+            if(product instanceof Aliment && product.isExpired()) {
```

Comment:
Note that in the previous comment I used the following branch condition:
```javascript
if ((product instanceof Aliment) && product.isExpired()) { }
```
Take care of the operator precedence. In this case, `&&` is evaluated before `instanceof` operator if we don't use brackets.

Take a look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#table